api/src/main/java/org/pmiops/workbench/db/dao/DataSetServiceImpl.java

```diff
@@ -676,6 +684,59 @@ public DbDataset cloneDataSetToWorkspace(
     return cohortDao.findAllByCohortIdIn(dataSetDao.findOne(dataSet.getDataSetId()).getCohortIds());
   }
 
+  @Override
+  public List<DbDataset> getDataSets(ResourceType resourceType, long resourceId) {
+    List<DbDataset> dbDataSets = new ArrayList<>();
+    if (ResourceType.COHORT.equals(resourceType)) {
+      dbDataSets = dataSetDao.findDataSetsByCohortIds(resourceId);
+    } else if (ResourceType.CONCEPT_SET.equals(resourceType)) {
+      dbDataSets = dataSetDao.findDataSetsByConceptSetIds(resourceId);
+    }
+    return dbDataSets;
+  }
+
+  @Override
+  public List<DbDataset> getInvalidDataSetsByWorkspace(DbWorkspace dbWorkspace) {
+
+    return dataSetDao.findByWorkspaceIdAndInvalid(dbWorkspace.getWorkspaceId(), false);
+  }
+
+  @Override
+  public void deleteDataSet(DbWorkspace dbWorkspace, Long dataSetId) {
+    DbDataset dataSet = getDbDataSet(dbWorkspace, dataSetId);
+    dataSetDao.delete(dataSet.getDataSetId());
+  }
+
+  @Override
+  public DbDataset getDbDataSet(DbWorkspace dbWorkspace, Long dataSetId) {
+
+    DbDataset dataSet = dataSetDao.findOne(dataSetId);
+    if (dataSet == null || dbWorkspace.getWorkspaceId() != dataSet.getWorkspaceId()) {
+      throw new NotFoundException(
+          String.format(
+              "No data set with ID %s in workspace %s.", dataSet, dbWorkspace.getFirecloudName()));
+    }
+    return dataSet;
+  }
+
+  @Override
+  public void markDirty(ResourceType resourceType, long resourceId) {
+    List<DbDataset> dbDataSetList = getDataSets(resourceType, resourceId);
+    dbDataSetList =
```

Comment:
If you can, it's better to avoid overwriting variables. I'd probably give the original set a different name, or just inline it so yo have 
```
getDataSets(resourceType, resourceId).stream()
            .map(
                dataSet -> {
                  dataSet.setInvalid(true);
                  return dataSet;
                })
            .collect(Collectors.toList());
```

But really I think you can just do `dbDataSetList.stream().forEach(ds -> ds.setInvalid(true));` to change them in place.